distributed/tests/test_client.py

```diff
@@ -6461,24 +6461,15 @@ async def f(stacklevel, mode=None):
     assert "cdn.bokeh.org" in data
 
 
-@gen_cluster(nthreads=[])
-async def test_client_gather_semaphore_loop(s):
-    async with Client(s.address, asynchronous=True) as c:
-        assert c._gather_semaphore._loop is c.loop.asyncio_loop
-
-
 @gen_cluster(client=True)
 async def test_as_completed_condition_loop(c, s, a, b):
     seq = c.map(inc, range(5))
     ac = as_completed(seq)
+    async for _ in ac:
+        pass
     assert ac.condition._loop == c.loop.asyncio_loop
 
 
-def test_client_connectionpool_semaphore_loop(s, a, b):
-    with Client(s["address"]) as c:
-        assert c.rpc.semaphore._loop is c.loop.asyncio_loop
```

Comment:
Is there a reason this test is dropped?

Reply:
the on py3.10+ `c.rpc.semaphore` doesn't bind to the eventloop until the semaphore is exhausted and `c.rpc.semaphore.acquire()` blocks so I decided it was too difficult to get the test to pass and so not worth the maintenance to keep it around as skipped on py3.10+ until 3.9 EOL